### ISSUE TITLE
Add ability to customize cache path and error message when not found

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/run-node#1]
Mainly so it can be reused in [`alfy`](https://github.com/sindresorhus/alfy), but others might have a need for this too. I'm thinking environment variables, but happy to consider better ideas.
## 

// @mafredri

**Comment by mafredri** [comment]
We could also use `getopts`, and provide this functionally through cli flags, e.g. `./run-node -c /cache/path -m 'Error message' index.js`

Although, in this case, providing flags to index.js would be risky and should perhaps be separated with a `--`, e.g. `./run-node -c /cache/path -m 'Error message' index.js -- -c -v -l`.

**Comment by sindresorhus** [comment]
That would conflict with `node` though, and even if we did `--` it could easily conflict some Node.js script reading those.

**Comment by mafredri** [comment]
I don't think it will conflict, since all parsed options would be removed, which would include `-c`, `-m` and `--`. So essentially my example above would call node in this way: `node index.js -c -v -l`.

But I think environment variables can also work, both have their own advantages / disadvantages. Environment variables are a bit harder, more hidden, to use and requires previous knowledge, whereas the cli options could also include `-h` and `--help`. Or actually, now that I think about it, a help in both cases would be ok.

**Comment by sindresorhus** [comment]
Ok, let me illustrate what I meant:

``` js
// x.js
console.log(process.argv.slice(2));
```

```
❯ node x.js -- -c -v -l
[ '--', '-c', '-v', '-l' ]
```

That means `x.js` would see those flags and might try to pass them to some other script that parses and uses it.

Using flags would also complicate the "proxy" I'll use in Alfy to defer to this script, as it needs to take out the `--` flags.

Just feels easier and safer with namespaced environment variables.

**Comment by mafredri** [comment]
Thanks for the illustration, I do get what you mean, node inheriting the arguments to `run-node`. But this is only the case if we pass `$@` along to the node binary. What I'm proposing is that we don't, this means that `-c /cache/path`, `-m errmsg` and `--` will never be sent as arguments to the node binary. And as in my example, `[ 'index.js', '-c', '-v', '-l' ]` is exactly what we intended to send.

> Using flags would also complicate the "proxy" I'll use in Alfy to defer to this script, as it needs to take out the -- flags.

It shouldn't need to concern itself with that at all!

> Just feels easier and safer with namespaced environment variables.

Like I said, I'm fine with that too :).

**Comment by sindresorhus** [comment]
> What I'm proposing is that we don't, this means that -c /cache/path, -m errmsg and -- will never be sent as arguments to the node binary.

Ok, let me turn it around then. What if the user actually wants to pass `-- --foo` to a module expecting it.

## 

Regardless, I think environment variables have clearer intent here.

## 

So to the option naming:
- `RUN_NODE_CACHE`
- `RUN_NODE_ERROR`

?

**Comment by mafredri** [comment]
> Ok, let me turn it around then. What if the user actually wants to pass -- --foo to a module expecting it.

`--` is common in unix and indicates that we stop processing args. So from there-on-out, everything will be as-is. This means that passing `--` is possible with `-- --`.

---

> So to the option naming:

Some minor suggestions for clarity:
- `RUN_NODE_CACHE_PATH` (if we expect a directory, maybe `FILE` if we expect a file?)
- `RUN_NODE_ERROR_MSG` (clearer)

**Comment by sindresorhus** [comment]
Sounds good to me. Thanks for the feedback. 👌😀

**Comment by sindresorhus** [comment]
This is now up for grabs if anyone want to work on it :)

**Comment by joaoppsantos** [comment]
Greetings
Made an implementation for this issue - can be found [here](https://github.com/sindresorhus/run-node/pull/3)

**Comment by sindresorhus** [comment]
Fixed by #4
